### PR TITLE
Regain previus caller position + Add support facilities for callback functionality

### PR DESCRIPTION
**Pull request by sip2thefuture**
This is for https://github.com/babblevoice/babble-sip/issues/10

### CODE REVIEW

File: lib/fifo.js [hunk 1]
```diff
@@ -187,6 +177,20 @@ class fifo {
...
 
+  /**
+  Get a call by callerid
+  */
+  getcallbycallerid( callerid ) {
+    for ( var i = 0; i < this._fifos.length; i++ ) {
+      const othercall = this._calls.get( this._fifos[i][0] )
+      if ( othercall
+          && othercall.call.callerid == othercall.call.callerid ) {
+        othercall.call.__position = i
```
Comment (tinpotnick) [inline lib/fifo.js]: minor tweak - I included the vars object in a call for anything outside of the call object to be able to play with - i.e. __postion could in the future, conflict with something in the call object in the future.

line 152 of fifo.js uses a variable: options.call.vars.fifo.position - which could be updated before returning it?

File: lib/fifo.js [hunk 2]
```diff
@@ -145,9 +145,15 @@ class fifo {
       this._pick.bind( this ) )
 
     this._callcount++
+
+    if ( options.positionstalecall ) {
```
Comment (tinpotnick) [inline lib/fifo.js]: I don't understand what this is for?

Comment (sip2thefuture) [reply lib/fifo.js]: Because we decided that it should hangup if the user didn't choose to cancel the callback unless I'm missing something

Comment (tinpotnick) [reply lib/fifo.js]: Would it be a cleaner interface to simply pick the call from babble-sip, i.e.

const c = queues.findbycallerid( { callerid: "0123" } ) 
if( c ) c.pick()

Comment (sip2thefuture) [reply lib/fifo.js]: are you implying implementing findbycallerid then?

Comment (sip2thefuture) [reply lib/fifo.js]: actually shouldn't be fine to hangup the call in queuing.js directly?

File: lib/fifo.js (hunk 1 above)
Comment (sip2thefuture) [reply lib/fifo.js]: this is now replaced by new code

File: lib/fifo.js (hunk 2 above)
Comment (sip2thefuture) [reply lib/fifo.js]: okay...gotcha I updated the code

**Review by tinpotnick — APPROVED** [review]

**Pull request merged**